ReactNativeClient/lib/reducer.js

```diff
@@ -652,7 +809,8 @@ const reducer = (state = defaultState, action) => {
 			break;
 
 		case 'FOLDER_DELETE':
-			newState = handleItemDelete(state, action);
+			newState = handleHistory(state, action);
+			newState = handleItemDelete(newState, action);
 			break;
 
 		case 'MASTERKEY_UPDATE_ALL':
```

Comment:
I think the better way to do this would be to convert handleHistory() into another reducer.

Then we can combine them both using combineReducers. Then we wouldn't have to change anything in the current reducer.